packages/algob/src/lib/tx.ts

```diff
@@ -96,208 +96,109 @@ export function encodeNote (note: string | undefined, noteb64: string| undefined
   return noteb64 ? encoder.encode(noteb64) : encoder.encode(note);
 }
 
-/**
-  * Description:
-  * This function is used to transfer Algos
-  * from one account to another using fromAccount secret key (for signing)
-
-  * Returns:
-  * Transaction details
-*/
-export async function transferMicroAlgos (
-  deployer: AlgobDeployer,
-  from: AccountSDK,
-  to: string,
-  amountMicroAlgos: number,
-  param: TxParams
-): Promise<tx.ConfirmedTxInfo> {
-  const params = await mkSuggestedParams(deployer.algodClient, param);
-
-  const receiver = to;
-  let note;
-  if (param.noteb64 ?? param.note) {
-    note = encodeNote(param.note, param.noteb64);
-  } else {
-    note = undefined;
+async function mkTransaction (deployer: AlgobDeployer, txnParam: execParams): Promise<Transaction> {
+  const params = await mkSuggestedParams(deployer.algodClient, txnParam.payFlags);
+  const note = encodeNote(txnParam.payFlags.note, txnParam.payFlags.noteb64);
+
+  switch (txnParam.type) {
+    case TransactionType.TransferAsset: {
+      return tx.makeAssetTransferTxnWithSuggestedParams(
+        txnParam.fromAccount.addr,
+        txnParam.toAccountAddr,
+        txnParam.payFlags.closeRemainderTo,
+        undefined,
+        txnParam.amount,
+        note,
+        txnParam.assetID,
+        params);
+    }
+    case TransactionType.TransferAlgo: {
+      return tx.makePaymentTxnWithSuggestedParams(
+        txnParam.fromAccount.addr,
+        txnParam.toAccountAddr,
+        txnParam.amountMicroAlgos,
+        txnParam.payFlags.closeRemainderTo,
+        note,
+        params);
+    }
+    case TransactionType.ClearSSC: {
+      return tx.makeApplicationClearStateTxn(txnParam.fromAccount.addr, params, txnParam.appId);
+    }
+    case TransactionType.DeleteSSC: {
+      return tx.makeApplicationDeleteTxn(txnParam.fromAccount.addr, params, txnParam.appId);
+    }
+    case TransactionType.CallNoOpSSC: {
+      return tx.makeApplicationNoOpTxn(
+        txnParam.fromAccount.addr,
+        params,
+        txnParam.appId,
+        txnParam.appArgs,
+        txnParam.accounts,
+        txnParam.foreignApps,
+        txnParam.foreignAssets,
+        note,
+        txnParam.lease,
+        txnParam.rekeyTo);
+    }
+    case TransactionType.CloseSSC: {
+      return tx.makeApplicationCloseOutTxn(txnParam.fromAccount.addr, params, txnParam.appId);
+    }
+    default: {
+      throw new Error("Unknown type of transaction");
```

Comment:
this has to be a `ALGOB` error. You can create a new error in `error-list`. It must take the `txnParam.type` as a parameter

Reply:
ok